arm/postgresql/models.go

```diff
@@ -210,10 +223,81 @@ type Server struct {
 
 // ServerForCreate is represents a server to be created.
 type ServerForCreate struct {
-	Sku        *Sku                       `json:"sku,omitempty"`
-	Properties *ServerPropertiesForCreate `json:"properties,omitempty"`
-	Location   *string                    `json:"location,omitempty"`
-	Tags       *map[string]*string        `json:"tags,omitempty"`
+	Sku        *Sku                      `json:"sku,omitempty"`
+	Properties ServerPropertiesForCreate `json:"properties,omitempty"`
+	Location   *string                   `json:"location,omitempty"`
+	Tags       *map[string]*string       `json:"tags,omitempty"`
+}
+
+// UnmarshalJSON is the custom unmarshaler for ServerForCreate model
+func (sfc *ServerForCreate) UnmarshalJSON(b []byte) error {
+	var m map[string]*json.RawMessage
+	err := json.Unmarshal(b, &m)
+	if err != nil {
+		return err
+	}
+
+	v := m["sku"]
+	if v != nil {
+		var sku Sku
+		err = json.Unmarshal(*m["sku"], &sku)
+		if err != nil {
+			return err
+		}
+		sfc.Sku = &sku
+	}
+
+	v = m["properties"]
+	if v != nil {
+		p, err := unmarshalIServerPropertiesForCreate(*m["properties"])
+		if err != nil {
+			return err
+		}
+		sfc.Properties = p
+	}
+
+	v = m["location"]
+	if v != nil {
+		var location string
+		err = json.Unmarshal(*m["location"], &location)
+		if err != nil {
+			return err
+		}
+		sfc.Location = &location
+	}
+
+	v = m["tags"]
+	if v != nil {
+		var tags map[string]*string
+		err = json.Unmarshal(*m["tags"], &tags)
+		if err != nil {
+			return err
+		}
+		sfc.Tags = &tags
+	}
+
+	return nil
+}
+
+func unmarshalIServerPropertiesForCreate(b []byte) (ServerPropertiesForCreate, error) {
```

Comment:
Shouldn't this be renamed to not include the "I" in the SerePropertiesForCreate?